src/libratbag.c

```diff
@@ -1090,14 +1090,16 @@ ratbag_resolution_set_dpi_xy(struct ratbag_resolution *resolution,
 {
 	struct ratbag_profile *profile = resolution->profile;
 
+	log_debug(profile->device->ratbag,"ratbag: profile %d resolution %d: set %dx%d dpi\n",profile->index,resolution->index,x,y);
+
 	if (!ratbag_resolution_has_capability(resolution,
 					      RATBAG_RESOLUTION_CAP_SEPARATE_XY_RESOLUTION))
 		return RATBAG_ERROR_CAPABILITY;
 
-	if ((x == 0 && y != 0) || (x != 0 && y == 0))
-		return RATBAG_ERROR_VALUE;
-
-	if (!resolution_has_dpi(resolution, x) || !resolution_has_dpi(resolution, y))
+	if (resolution->is_disabled) {
+		if (!ratbag_resolution_has_capability(resolution, RATBAG_RESOLUTION_CAP_DISABLE))
+			return RATBAG_ERROR_CAPABILITY;
+	} else if (!resolution_has_dpi(resolution, x) || !resolution_has_dpi(resolution, y))
```

Comment:
Please keep the original check (`(x == 0 && y != 0) || (x != 0 && y == 0)`), it is fine for a resolution to have `dpi_x` but no `dpi_y` and vice-versa.

Reply:
Very strange scenario but I can adjust it.

Reply:
I agree, but it could be possible, even if very very unlikely, and it does not really hurt to have the proper check.